nikola/plugins/task/galleries.py

```diff
@@ -229,7 +236,18 @@ def gen_tasks(self):
 
                 image_name_list = [os.path.basename(p) for p in image_list]
 
-                if self.kw['use_filename_as_title']:
+                if self.captions:
+                    img_titles = []
+                    for fn in image_name_list:
+                        if fn in self.captions:
+                            img_titles.append(self.captions[fn])
+                        else:
+                            img_titles.append(fn)
+                            self.logger.warn(
```

Comment:
No need for a warning IMO.

Reply:
Made this a self.logger.debug() call instead.